src/error.rs

```diff
@@ -26,7 +26,9 @@ pub enum Error {
     GroupsError(GroupsValidationError),
     GroupsUpconversionError(GroupsValidationError),
     ExpectedPlistDictionaryError,
+    ExpectedPlistStringError,
     ExpectedPositiveValue,
+    TypeError(ErrorKind),
```

Comment:
I would call this a `ParseError` maybe? Or an `InvalidData` error? `ErrorKind` describes, (usually) issues we encounter while parsing a `.glif`.